sys/mips/malta.c

```diff
@@ -122,6 +125,19 @@ char *kenv_get(const char *key) {
   return NULL;
 }
 
+char **kenv_get_init_args(int *n) {
+  for (int i = 1; i < _kenv.argc; i++) {
+    char *arg = _kenv.argv[i];
+    if ((strncmp("--", arg, 2) == 0)) {
+      if (n) {
+        *n = _kenv.argc - i - 1;
+      }
+      return _kenv.argv + i + 1;
+    }
+  }
+  return NULL;
+}
```

Comment:
An idea: changing `return NULL;` to `*n = 0; return NULL;` would enable you to merge two cases in file `sys/kern/main.c`: `if (init_args) { ... } else { ... }` into just one. Then you could delete:
 ```
run_program(init, (char *[]){init, NULL}, (char *[]){NULL});
```
and always call:
```
run_program(init, args, (char *[]){NULL});
```

Reply:
Done. (d6fd9d470780afc6d27bc4f28035418f04d7a828)